### PR TITLE
Add class examples for the documentation

### PR DESCRIPTION
**Pull request by angranl-flex**
For now, the class examples are added for all sections except Meshing

### CODE REVIEW

File: flow360/component/simulation/models/volume_models.py [hunk 1]
```diff
@@ -315,8 +481,33 @@ class BETDisk(Flow360BaseModel):
     in :ref:`Case Studies <XV15BETDisk_caseStudy>`.
     Because a transient BET Line simulation is simply a time-accurate version of a steady-state
     BET Disk simulation, most of the parameters below are applicable to both methods.
-    Note that `center_of_rotation`, `axis_of_rotation`, `radius`, `thickness` can be acquired from `entity`
-    so they are not required anymore.
+
+    Note
+    ----
+    1. :py:attr:`BETDisk.entities` needs to be defined using :class:`Cylinder`.
```
Comment (benflexcompute) [inline flow360/component/simulation/models/volume_models.py:487]: Again there is no need. It is shown as field definition.

File: flow360/component/simulation/outputs/output_entities.py [hunk 2]
```diff
@@ -43,7 +43,23 @@ class _PointEntityBase(EntityBase, metaclass=ABCMeta):
...
 class Slice(_SliceEntityBase):
-    """:class:`Slice` class for defining a slice for :class:`~flow360.SliceOutput`."""
+    """
+
+    :class:`Slice` class for defining a slice for :class:`~flow360.SliceOutput`.
+
+    Example
+    -------
+
+    Define a :class:`Slice` along (0,1,0) direction with the origin of (0,2,0).
```
Comment (benflexcompute) [inline flow360/component/simulation/outputs/output_entities.py:53]:
```suggestion
    Define a :class:`Slice` with normal direction (0,1,0) with the origin of (0,2,0) fl.u.m.
```

File: flow360/component/simulation/outputs/output_entities.py [hunk 3]
```diff
@@ -53,7 +69,23 @@ class Slice(_SliceEntityBase):
...
 class Isosurface(_OutputItemBase):
-    """:class:`Isosurface` class for defining an isosurface for :class:`~flow360.IsosurfaceOutput`."""
+    """
+
+    :class:`Isosurface` class for defining an isosurface for :class:`~flow360.IsosurfaceOutput`.
+
+    Example
+    -------
+
+    Define a :class:`Isosurface` with the :py:attr:`iso_value` of :code:`T` to be 1.5.
```
Comment (benflexcompute) [inline flow360/component/simulation/outputs/output_entities.py:79]:
```suggestion
    Define a :class:`Isosurface` of temperature equal to 1.5 non-dimensional temperature.
```

File: flow360/component/simulation/outputs/output_entities.py [hunk 4]
```diff
@@ -79,8 +121,22 @@ class Point(_PointEntityBase):
 
 class PointArray(_PointEntityBase):
     """
-    :class:`PointArray` class for defining a line for
+    :class:`PointArray` class for defining multiple monitor points along a line for
```
Comment (benflexcompute) [inline flow360/component/simulation/outputs/output_entities.py:124]:
```suggestion
    :class:`PointArray` class for defining multiple equally spaced monitor points along a line for
```

File: flow360/component/simulation/outputs/output_entities.py [hunk 5]
```diff
@@ -79,8 +121,22 @@ class Point(_PointEntityBase):
...
     :class:`~flow360.ProbeOutput`/:class:`~flow360.SurfaceProbeOutput`.
+
+    Example
+    -------
+    Define :class:`PointArray` with 6 points along a line starting from (0,0,0) to (1,2,3).
```
Comment (benflexcompute) [inline flow360/component/simulation/outputs/output_entities.py:129]:
```suggestion
    Define :class:`PointArray` with 6 equally spaced points along a line starting from (0,0,0) to (1,2,3).
```

File: flow360/component/simulation/outputs/outputs.py [hunk 6]
```diff
@@ -33,7 +33,38 @@
...
+    ...     + " / sqrt(gamma * primitiveVars[4] / primitiveVars[0]);",
+    ... )
+
+
+    - Compute :code:`PressureForce` using :class:`UserDefinedField`:
+
+    >>> fl.UserDefinedField(
+    ...     name="PressureForce",
+    ...     expression="double p = primitiveVars[4] - pressureFreestream; "
+    ...     + "PressureForce[0] = p * rotationModel_nodeNormals[0]; "
```
Comment (benflexcompute) [inline flow360/component/simulation/outputs/outputs.py:60]: Are you sure `rotationModel_nodeNormals` is correct?

File: flow360/component/simulation/outputs/outputs.py [hunk 7]
```diff
@@ -100,14 +157,37 @@ class SurfaceOutput(_AnimationAndFileFormatSettings):
...
+
+    Warnings
+    --------
+
+    The solver only accepts exactly the same set of :py:attr:`output_fields`
+    (is shared) between :class:`SurfaceOutput` and :class:`TimeAverageSurfaceOutput`.
+
+    Note
+    ----
+    Old `computeTimeAverages` can be infered when user is explicitly using for
```
Comment (benflexcompute) [inline flow360/component/simulation/outputs/outputs.py:169]: No need for migration hints.

File: flow360/component/simulation/outputs/outputs.py [hunk 8]
```diff
@@ -100,14 +157,37 @@ class SurfaceOutput(_AnimationAndFileFormatSettings):
...
+    example :class:`TimeAverageSurfaceOutput`.
+
+    Example
+    -------
+
+    Calculate the average value starting from the :math:`4^{th}` physical step.
+    The results are output every 10 physical step starting from the :math:`14^{th}` physical step.
```
Comment (benflexcompute) [inline flow360/component/simulation/outputs/outputs.py:176]:
```suggestion
    The results are output every 10 physical step starting from the :math:`14^{th}` physical step (14, 24, 34 etc).
```

File: flow360/component/simulation/outputs/outputs.py [hunk 9]
```diff
@@ -133,15 +225,34 @@ class VolumeOutput(_AnimationAndFileFormatSettings):
...
     between :class:`VolumeOutput` and :class:`TimeAverageVolumeOutput`.
     Also let's not worry about allowing entities here as it is not supported by solver anyway.
 
-    Notes
-    -----
-        Old `computeTimeAverages` can be infered when user is explicitly using for example
-        `TimeAverageSurfaceOutput`.
+    Note
+    ----
+    Old `computeTimeAverages` can be infered when user is explicitly using for example
```
Comment (benflexcompute) [inline flow360/component/simulation/outputs/outputs.py:238]: Remove

File: flow360/component/simulation/outputs/outputs.py [hunk 10]
```diff
@@ -133,15 +225,34 @@ class VolumeOutput(_AnimationAndFileFormatSettings):
...
+    :class:`TimeAverageVolumeOutput`.
+
+    Example
+    -------
+
+    Calculate the average value starting from the :math:`4^{th}` physical step.
+    The results are output every 10 physical step starting from the :math:`14^{th}` physical step.
```
Comment (benflexcompute) [inline flow360/component/simulation/outputs/outputs.py:245]: List steps

File: flow360/component/simulation/outputs/outputs.py [hunk 11]
```diff
@@ -178,7 +332,36 @@ class TimeAverageSliceOutput(SliceOutput):
...
+    """
+
+    :class:`IsosurfaceOutput` class for isosurface output settings.
+
+    Example
+    -------
+
+    Define the :class:`IsosurfaceOutput` of :code:`qcriterion` on two isosurfaces:
+
+    - :code:`Isosurface_T_0.1` has the :py:attr:`~Isosurface.iso_value` of :code:`T` to be 0.1;
```
Comment (benflexcompute) [inline flow360/component/simulation/outputs/outputs.py:344]: Mention it is nondimensional.

File: flow360/component/simulation/outputs/outputs.py [hunk 12]
```diff
@@ -239,8 +485,56 @@ def check_unique_probe_type(cls, value):
 
 class SurfaceProbeOutput(Flow360BaseModel):
     """
-    :class:`SurfaceProbeOutput` class for surface probe monitor output settings.
-    The monitor location will be projected to the surface closest to the point.
+    :class:`SurfaceProbeOutput` class for setting surface output data probed at monitor points.
+    The specified monitor point will be projected to the :py:attr:`~SurfaceProbeOutput.target_surfaces`
+    closest to the point. The probed results on the projected point is output.
```
Comment (benflexcompute) [inline flow360/component/simulation/outputs/outputs.py:490]:
```suggestion
    closest to the point. The probed results on the projected point will be dumped.
```

File: flow360/component/simulation/primitives.py [hunk 13]
```diff
@@ -265,6 +279,15 @@ def from_principal_axes(
     ):
         """
         Construct box from principal axes
+
```
Comment (benflexcompute) [inline flow360/component/simulation/primitives.py:282]: Can you move this to the Box docstring?

File: flow360/component/simulation/user_defined_dynamics/user_defined_dynamics.py [hunk 14]
```diff
@@ -11,7 +11,41 @@
 
 
 class UserDefinedDynamic(Flow360BaseModel):
-    """:class:`UserDefinedDynamic` class for defining the user defined dynamics inputs."""
+    """
+    :class:`UserDefinedDynamic` class for defining the user defined dynamics inputs.
+
+    Example
+    -------
+
```
Comment (benflexcompute) [inline flow360/component/simulation/user_defined_dynamics/user_defined_dynamics.py:19]: Can you add link to where this comes from?

File: flow360/component/simulation/primitives.py (hunk 13 above)
Comment (angranl-flex) [reply flow360/component/simulation/primitives.py:282]: Sure, updated

File: flow360/component/simulation/user_defined_dynamics/user_defined_dynamics.py (hunk 14 above)
Comment (angranl-flex) [reply flow360/component/simulation/user_defined_dynamics/user_defined_dynamics.py:19]: Yes, add the link to the corresponding tutorial page

File: flow360/component/simulation/outputs/outputs.py (hunk 12 above)
Comment (angranl-flex) [reply flow360/component/simulation/outputs/outputs.py:490]: Updated

File: flow360/component/simulation/outputs/outputs.py (hunk 11 above)
Comment (angranl-flex) [reply flow360/component/simulation/outputs/outputs.py:344]: OK, updated

File: flow360/component/simulation/outputs/outputs.py (hunk 9 above)
Comment (angranl-flex) [reply flow360/component/simulation/outputs/outputs.py]: Removed

File: flow360/component/simulation/outputs/outputs.py (hunk 10 above)
Comment (angranl-flex) [reply flow360/component/simulation/outputs/outputs.py]: Yes, list the steps

File: flow360/component/simulation/outputs/outputs.py (hunk 7 above)
Comment (angranl-flex) [reply flow360/component/simulation/outputs/outputs.py]: Removed

File: flow360/component/simulation/outputs/outputs.py (hunk 8 above)
Comment (angranl-flex) [reply flow360/component/simulation/outputs/outputs.py]: Accepted

File: flow360/component/simulation/models/volume_models.py [hunk 15]
```diff
@@ -100,7 +130,19 @@ class NavierStokesInitialCondition(ExpressionInitialConditionBase):
     """
     :class:`NavierStokesInitialCondition` class for specifying the
     :py:attr:`Fluid.initial_condition`.
-    By default
+
```
[the remark this answers is not in the document]
Comment (angranl-flex) [reply flow360/component/simulation/models/volume_models.py:133]: Sure, updated

File: flow360/component/simulation/outputs/outputs.py (hunk 6 above)
Comment (angranl-flex) [reply flow360/component/simulation/outputs/outputs.py]: Fixed

File: flow360/component/simulation/outputs/outputs.py [hunk 16]
```diff
@@ -210,7 +389,55 @@ class SurfaceIntegralOutput(Flow360BaseModel):
...
+    ...             location=(0.0, -1.5, 0.0) * fl.u.m,
+    ...         ),
+    ...     ],
+    ...     output_fields=["primitiveVars"],
+    ... )
+
+
+    - Define :class:`ProbeOutput` on monitor points along the line.
+      :code:`Line_1` is from (1,0,0) to (1.5,0,0) and has 6 monitor points.
+      :code:`Line_2` is from (-1,0,0) to (-1.5,0,0) and has 3 monitor points.
```
Comment (benflexcompute) [inline flow360/component/simulation/outputs/outputs.py:418]: Can you list the points explicitly here? You can change -1.5 to for example -1.9 so the coordinates have finite digits.

Comment (benflexcompute) [reply flow360/component/simulation/outputs/outputs.py:418]: Also add units here.

File: flow360/component/simulation/models/surface_models.py [hunk 17]
```diff
@@ -151,13 +194,44 @@ class Rotational(Flow360BaseModel):
...
+    - Define isoflux wall boundary condition on entities
+      with the naming pattern :code:`"solid/isoflux-*"`:
+
+    >>> fl.Wall(
+    ...     entities=volume_mesh["solid/isoflux-*"],
+    ...     heat_spec=fl.HeatFlux(1.0 * fl.u.W/fl.u.m**2),
+    ... )
+
+
+    - Define isothermal wall boundary condition on entities
```
Comment (benflexcompute) [inline flow360/component/simulation/models/surface_models.py:226]: Duplicate example. Remove

**Review by benflexcompute — APPROVED** [review]

File: flow360/component/simulation/models/surface_models.py (hunk 17 above)
Comment (angranl-flex) [reply flow360/component/simulation/models/surface_models.py]: Updated

**Pull request merged**